prompt_toolkit/mouse_events.py

```diff
@@ -19,15 +19,44 @@
 
 from .data_structures import Point
 
-__all__ = ["MouseEventType", "MouseEvent"]
+__all__ = ["MouseEventType", "MouseButton", "MouseModifier", "MouseEvent"]
 
 
+# fmt: off
 class MouseEventType(Enum):
-    MOUSE_UP = "MOUSE_UP"
-    MOUSE_DOWN = "MOUSE_DOWN"
-    MOUSE_DOWN_MOVE = "MOUSE_DOWN_MOVE"
-    SCROLL_UP = "SCROLL_UP"
+    MOUSE_UP    = "MOUSE_UP"    #Ryan Burgert: This same event type is fired for all three events: left mouse up, right mouse up, or middle mouse up
+    MOUSE_DOWN  = "MOUSE_DOWN"  #Ryan Burgert: This implicitly refers to the left mouse down (this event is not fired upon pressing the middle or right mouse buttons. I didn't change it's name because I fear creating incompatiabilities with older code.
+    SCROLL_UP   = "SCROLL_UP"
     SCROLL_DOWN = "SCROLL_DOWN"
+    MOUSE_DRAG  = "MOUSE_DRAG"  #Triggered when the left   mouse button is held down, and the mouse moves
+
+
+class MouseButton(Enum):
+    LEFT           = "LEFT"
+    MIDDLE         = "MIDDLE"
+    RIGHT          = "RIGHT"
+    NO_BUTTON      = ""                # When we're scrolling, or just moving the mouse and not pressing a button, mouse_event.button=="". The reason it's an empty string is so that bool(MouseButton.NO_BUTTON)==False
```

Comment:
Can you put the comments above the constants, and wrap them at 80 characters?

Reply:
I don't like that people would rely on ``bool(MouseButton.NO_BUTTON)==False``. I don't think mypy would approve it either. Can you make it ``NO_BUTTON = "NO_BUTTON"``?

Reply:
Done